chain-evm/src/state/account.rs

```diff
@@ -68,19 +72,28 @@ pub type ByteCode = Vec<u8>;
 /// A represantation of an EVM account.
 #[derive(Clone, Default, PartialEq, Eq)]
 pub struct Account {
-    /// Account nonce. A number of value transfers from this account.
-    pub nonce: Nonce,
```

Comment:
Not that it's crucial, but I would like to understand the reasoning behind nonce and balance being split in different structs, is it because of the shared balance stuff?

Reply:
Yes, you are correct, `AccountState` also used for the jormungandr `AccountState<Extra>` as a field